packages/svelte/src/internal/client/runtime.js

```diff
@@ -1298,6 +1298,20 @@ export function pre_effect(init) {
 	);
 }
 
+/**
+ * @param {() => void | (() => void)} init
+ * @returns {import('./types.js').EffectSignal}
+ */
+export function invalidate_effect(init) {
```

Comment:
Should start adding doc comments for what each of these effects are, when they run, what they are intended for etc.